org.lflang/src/org/lflang/generator/c/CCoreFilesUtils.java

```diff
@@ -111,14 +112,15 @@ private static List<String> getThreadSupportFiles(
         boolean threading,
         SchedulerOption scheduler
     ) {
-        return threading ?
-                List.of(
-                    "threaded/scheduler.h",
-                    "threaded/scheduler_instance.h",
-                    "threaded/scheduler_sync_tag_advance.c",
-                    "threaded/scheduler_" + scheduler + ".c",
-                    "threaded/reactor_threaded.c"
-                ) :
-                List.of("reactor.c");
+        List<String> supportFiles = new ArrayList<String>(Arrays.asList(
+            "threaded/scheduler.h",
+            "threaded/scheduler_instance.h",
+            "threaded/scheduler_sync_tag_advance.c",
+            "threaded/scheduler_" + scheduler + ".c",
+            "threaded/reactor_threaded.c"
+        ));
+        if (scheduler == SchedulerOption.QS)
```

Comment:
This would also be better handled [here](https://github.com/lf-lang/lingua-franca/blob/d561add3e18a2e65b8648f240a4c7609819eb653/org.lflang/src/org/lflang/generator/c/CGenerator.java#L864).